src/java.net.http/share/classes/jdk/internal/net/http/common/Utils.java

```diff
@@ -912,107 +907,72 @@ public static String hostString(HttpRequestImpl request) {
+    public static Logger getHpackLogger(Supplier<String> dbgTag) {
+        return DebugLogger.createHpackLogger(dbgTag, DEBUG_HPACK_CONFIG);
     }
 
     /**
      * Get a logger for debug WebSocket traces.The logger should only be used
      * with levels whose severity is {@code <= DEBUG}.
+     * <p>
+     * The logger should only be used with levels whose severity is
+     * {@code <= DEBUG}.
+     * <p>
+     * The output of this logger is controlled by the system property
+     * -Djdk.internal.httpclient.websocket.debug. The value of the property is
+     * a comma separated list of tokens. The following tokens are
+     * recognized:
+     * <ul>
+     *   <li> err: the messages will be logged on System.err</li>
+     *   <li> out: the messages will be logged on System.out</li>
+     *   <li> log: the messages will be forwarded to an internal
+     *        System.Logger named "jdk.internal.httpclient.websocket.debug"</li>
+     *   <li> true: this is equivalent to "err,log":  the messages will be logged
+     *        both on System.err, and forwarded to the internal logger.</li>
+     * </ul>
      *
-     * By default, this logger will forward all messages logged to an internal
-     * logger named "jdk.internal.httpclient.websocket.debug".
-     * In addition, the provided boolean {@code on==true}, it will print the
-     * messages on stderr.
-     * The logger will add some decoration to the printed message, in the form of
+     * This logger will add some decoration to the printed message, in the form of
      * {@code <Level>:[<thread-name>] [<elapsed-time>] <dbgTag>: <formatted message>}
      *
-     * @apiNote To obtain a logger that will always print things on stderr in
-     *          addition to forwarding to the internal logger, use
-     *          {@code getWebSocketLogger(this::dbgTag, true);}.
-     *          This is also equivalent to calling
-     *          {@code getWebSocketLogger(this::dbgTag, Level.ALL);}.
-     *          To obtain a logger that will only forward to the internal logger,
-     *          use {@code getWebSocketLogger(this::dbgTag, false);}.
-     *          This is also equivalent to calling
-     *          {@code getHpackLogger(this::dbgTag, Level.OFF);}.
      *
      * @param dbgTag A lambda that returns a string that identifies the caller
      *               (e.g: "WebSocket(3)")
-     * @param on  Whether messages should also be printed on
-     *            stderr (in addition to be forwarded to the internal logger).
      *
      * @return A logger for WebSocket internal debug traces
      */
-    public static Logger getWebSocketLogger(Supplier<String> dbgTag, boolean on) {
-        Level errLevel = on ? Level.ALL : Level.OFF;
-        return getWebSocketLogger(dbgTag, errLevel);
+    public static Logger getWebSocketLogger(Supplier<String> dbgTag) {
+        return DebugLogger.createWebSocketLogger(dbgTag, DEBUG_CONFIG);
```

Comment:
I think this should should be:

```
return DebugLogger.createWebSocketLogger(dbgTag, DEBUG_WS_CONFIG);
```
i.e. `DEBUG_WS_CONFIG` instead of `DEBUG_CONFIG`

Reply:
Good catch!